SDK Binaries moved to SDK-Version dependend folder - build failes currently

On a clean installation of Windows 10 1803 + VS 2017 15.7.1 + Windows 10 SDK (10.0.17134.0, only installed sdk) there are no binaries anymore in $(MSBuildProgramFiles32)\Windows Kits\10\bin\x64\

![image](https://user-images.githubusercontent.com/859801/40006946-e4269c12-579b-11e8-9c90-a7769952c3be.png)

They moved to sdk- versioned folders (C:\Program Files (x86)\Windows Kits\10\bin\10.0.17134.0\x64):

![image](https://user-images.githubusercontent.com/859801/40006981-fbb194d6-579b-11e8-9042-17af40e4ff97.png)


So the PrebuildScript failes https://github.com/onovotny/SignService/blob/master/src/SignService/SignService.csproj#L45

Error output:

> 2>SignClient -> C:\Users\awi.TECHTALK\Source\Repos\Azure.SignService\src\SignClient\bin\Debug\netcoreapp2.0\SignClient.dll
> 1>SignService -> C:\Users\awi.TECHTALK\Source\Repos\Azure.SignService\src\SignService\bin\Debug\netcoreapp2.0\SignService.dll
> 1>C:\Users\awi.TECHTALK\Source\Repos\Azure.SignService\src\SignService\SignService.csproj(45,5): error MSB3030: Could not copy the file "C:\Program Files (x86)\Windows Kits\10\bin\x64\appxpackaging.dll" because it was not found.
> 1>C:\Users\awi.TECHTALK\Source\Repos\Azure.SignService\src\SignService\SignService.csproj(45,5): error MSB3030: Could not copy the file "C:\Program Files (x86)\Windows Kits\10\bin\x64\opcservices.dll" because it was not found.
> 1>C:\Users\awi.TECHTALK\Source\Repos\Azure.SignService\src\SignService\SignService.csproj(45,5): error MSB3030: Could not copy the file "C:\Program Files (x86)\Windows Kits\10\bin\x64\Microsoft.Windows.Build.Appx.AppxPackaging.dll.manifest" because it was not found.
> 1>C:\Users\awi.TECHTALK\Source\Repos\Azure.SignService\src\SignService\SignService.csproj(45,5): error MSB3030: Could not copy the file "C:\Program Files (x86)\Windows Kits\10\bin\x64\Microsoft.Windows.Build.Appx.OpcServices.dll.manifest" because it was not found.
> 1>C:\Users\awi.TECHTALK\Source\Repos\Azure.SignService\src\SignService\SignService.csproj(45,5): error MSB3030: Could not copy the file "C:\Program Files (x86)\Windows Kits\10\bin\x64\makeappx.exe" because it was not found.
> 1>C:\Users\awi.TECHTALK\Source\Repos\Azure.SignService\src\SignService\SignService.csproj(45,5): error MSB3030: Could not copy the file "C:\Program Files (x86)\Windows Kits\10\bin\x64\makepri.exe" because it was not found.
> 1>C:\Users\awi.TECHTALK\Source\Repos\Azure.SignService\src\SignService\SignService.csproj(45,5): error MSB3030: Could not copy the file "C:\Program Files (x86)\Microsoft SDKs\Windows\v10.0A\bin\NETFX 4.7 Tools\mage.exe" because it was not found.
> 1>Done building project "SignService.csproj" -- FAILED.


Proposed solution for this issue:

I would add probing for existence of one of the files in the different sdk folders, starting with the latest version first.
Problem is only, for a new SDK the csproj has to be adjusted.

Would be this fine?

--- a/src/SignService/SignService.csproj
+++ b/src/SignService/SignService.csproj
@@ -3,7 +3,20 @@
   <PropertyGroup>
     <TargetFramework>netcoreapp2.0</TargetFramework>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
-    <WinSdkBinDir Condition="'$(WinSdkBinDir)' == ''">$(MSBuildProgramFiles32)\Windows Kits\10\bin\x64\</WinSdkBinDir>
+    
+    <WinSdk17134BinDir>$(MSBuildProgramFiles32)\Windows Kits\10\bin\10.0.17134.0\x64\</WinSdk17134BinDir>
+    <WinSdk16299BinDir>$(MSBuildProgramFiles32)\Windows Kits\10\bin\10.0.16299.0\x64\</WinSdk16299BinDir>
+    <WinSdk15063BinDir>$(MSBuildProgramFiles32)\Windows Kits\10\bin\10.0.15063.0\x64\</WinSdk15063BinDir>
+    <WinSdk14393BinDir>$(MSBuildProgramFiles32)\Windows Kits\10\bin\10.0.14393.0\x64\</WinSdk14393BinDir>
+
+    <ProbingFile>makeappx.exe</ProbingFile>
+    
+    <WinSdkBinDir Condition="'$(WinSdkBinDir)' == '' and Exists('$(WinSdk17134BinDir)$(ProbingFile)')">$(WinSdk17134BinDir)</WinSdkBinDir>
+    <WinSdkBinDir Condition="'$(WinSdkBinDir)' == '' and Exists('$(WinSdk16299BinDir)$(ProbingFile)')">$(WinSdk16299BinDir)</WinSdkBinDir>
+    <WinSdkBinDir Condition="'$(WinSdkBinDir)' == '' and Exists('$(WinSdk15063BinDir)$(ProbingFile)')">$(WinSdk15063BinDir)</WinSdkBinDir>
+    <WinSdkBinDir Condition="'$(WinSdkBinDir)' == '' and Exists('$(WinSdk14393BinDir)$(ProbingFile)')">$(WinSdk14393BinDir)</WinSdkBinDir>
+    
+    
     <NetSdkBinDir Condition="'$(NetSdkBinDir)' == ''">$(MSBuildProgramFiles32)\Microsoft SDKs\Windows\v10.0A\bin\NETFX 4.7 Tools\</NetSdkBinDir>
     <DockerComposeProjectPath>..\..\docker\docker-compose.dcproj</DockerComposeProjectPath>
     <UserSecretsId>9c2a9e27-e4ca-46ef-a555-2b557d723c4d</UserSecretsId>
@@ -41,8 +54,12 @@
     <DotNetCliToolReference Include="Microsoft.Extensions.SecretManager.Tools" Version="2.0.0" />
   </ItemGroup>
 
-  <Target Name="PrebuildScript" BeforeTargets="CoreBuild">
+  <Target Name="PrebuildScript" BeforeTargets="CoreBuild" Condition="'$(WinSdkBinDir)' != ''">
     <Copy SourceFiles="@(SdkFile)" DestinationFolder="tools\SDK" SkipUnchangedFiles="true" />
+  </Target>
+
+  <Target Name="PrebuildScript" BeforeTargets="CoreBuild" Condition="'$(WinSdkBinDir)' == ''">
+    <Error Text="No supported WinSDK could be found!" />
   </Target>
 
   <ItemGroup>

